include/dali/core/error_handling.h

```diff
@@ -234,28 +248,28 @@ inline dali::string GetStacktrace() {
 #define FILE_AND_LINE __FILE__ ":" DALI_STR(__LINE__)
 
 #define DALI_MESSAGE_WITH_STACKTRACE(str)\
-  (std::string("[" FILE_AND_LINE "] ") + str + dali::GetStacktrace())
+  (std::string("[" FILE_AND_LINE "] ")), (std::string(str)), dali::GetStacktrace()
 
 #define DALI_MESSAGE(str)\
-  (std::string("[" FILE_AND_LINE "] ") + str)
+  (std::string("[" FILE_AND_LINE "] ")), (std::string(str))
 
 #define DALI_FAIL(str)                            \
     throw dali::DALIException(DALI_MESSAGE_WITH_STACKTRACE(str));
 
-#define DALI_ERROR(str)                                          \
-  do {                                                           \
-    std::cerr << DALI_MESSAGE_WITH_STACKTRACE(str) << std::endl; \
+#define DALI_ERROR(str)                                                             \
+  do {                                                                              \
+    std::cerr << dali::make_string(DALI_MESSAGE_WITH_STACKTRACE(str)) << std::endl; \
   } while (0)
 
-#define DALI_WARN(...)                                                      \
-  do {                                                                      \
-    std::cerr << DALI_MESSAGE(dali::make_string(__VA_ARGS__)) << std::endl; \
+#define DALI_WARN(str)                                              \
+  do {                                                              \
+    std::cerr << dali::make_string(DALI_MESSAGE(str)) << std::endl; \
   } while (0)
 
-#define DALI_WARN_ONCE(...)                                                          \
-  do {                                                                               \
-    static int dummy =                                                               \
-        (std::cerr << DALI_MESSAGE(dali::make_string(__VA_ARGS__)) << std::endl, 0); \
+#define DALI_WARN_ONCE(str)                                                  \
+  do {                                                                       \
+    static int dummy =                                                       \
+        (std::cerr << dali::make_string(DALI_MESSAGE(str)) << std::endl, 0); \
```

Comment:
Now that's clever :) Still, we don't need to produce a string to print a message.
```suggestion
    static int dummy =                                                       \
        (print(std::cerr, DALI_MESSAGE(str), std::endl), 0); \
```